client/main.lua

```diff
@@ -2,6 +2,9 @@ function HUD:Toggle(state)
     SendNUIMessage({ type = "SHOW", value = state })
 end
 
+RegisterNetEvent('esx_hud:HudToggle', HUD:Toggle)
```

Comment:
```lua
RegisterCommand("togglehud", function()
    HUD:Toggle(HUD.Data.hudHidden)
    HUD.Data.hudHidden = not HUD.Data.hudHidden --> you should add this line to export
end, false)
```
So you should make a wrapper method where you type HUD:Toggle and then change the hudHidden state, so the logic is fully functional.

```suggestion

function ToggleHud(state)
    HUD:Toggle(state)
    HUD.Data.hudHidden = not state
end

RegisterNetEvent('esx_hud:HudToggle', ToggleHud)
```